pkgs/development/compilers/bs-platform/bs-platform-62.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchgit, ninja, nodejs, python35 }:
+let
+  version = "6.2.1";
+  ocaml-version = "4.06.1";
+  src = fetchgit {
+    url = "https://github.com/BuckleScript/bucklescript";
+    rev = "59357b7a2a5063718a385438e5cb0de7bd3b5398";
+    sha256 = "03qcdgqqybxlby6drmgnjx48mp4c1i2nxihqj3qfclkr2lmk0i1m";
```

Comment:
As per CI the actual sha seems to vary, locally I got:
```
hash mismatch in fixed-output derivation '/nix/store/xgyxf0d20cif6wkx1ia3vw2v3kh3i77y-bucklescript-59357b7':
  wanted: sha256:03qcdgqqybxlby6drmgnjx48mp4c1i2nxihqj3qfclkr2lmk0i1m
  got:    sha256:0adc20mp74znfvjb3n9whmn66waf6c0ddminkr9spssda3m4z58k
cannot build derivation '/nix/store/17b154nfk6al2pvw7kafmcd37fz0qsqv-bs-platform-6.2.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/17b154nfk6al2pvw7kafmcd37fz0qsqv-bs-platform-6.2.1.drv' failed
```
Perhaps some complexity around git submodules.

Reply:
Would be interested to know if it's possible to install from the tar archive e.g. https://github.com/BuckleScript/bucklescript/archive/6.2.1.tar.gz

Reply:
unfortunetely `fetchurl` on that tarball isn't pulling submodules. Anyway I think you're probably right that there are some problems with sha - One person reported some issues to me via email.

Reply:
Ah, possibly we want `fetchSubmodules` instead? Could try:
```nix
fetchFromGitHub {
  owner = "BuckleScript";
  repo = "bucklescript";
  rev = "${version}"; #= 6.2.1
  sha256 = "0zx9nq7cik0c60n3rndqfqy3vdbj5lcrx6zcqcz2d60jjxi1z32y";
  fetchSubmodules = true;
}
```

Reply:
@gamb that's what I'm currently doing.... compiling still..

Reply:
@turboMaCk the above builds for me, fyi :+1:

Reply:
It builds for me as well! Thanks!

Reply:
if you look through the fetchFromGitHub helper, `fetchSubmodules = true;` will make it call fetchgit. 

Just FYI :)

Reply:
interesting. Anyway I have ti confirmed now that this resolves the issue even for the user who reported it via email.